src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
```

Comment:
The condition `a <= 0 || b <= 0 || c <= 0` is correct for checking non-positive side lengths. However, ensure that the variables `a`, `b`, and `c` are correctly initialized in the constructor, as they are not declared as class properties.